tools/run_tests/artifact_targets.py

```diff
@@ -175,7 +175,8 @@ def build_jobspec(self):
     else:
       environ = {'CONFIG': 'opt',
                  'EMBED_OPENSSL': 'true',
-                 'EMBED_ZLIB': 'true'}
+                 'EMBED_ZLIB': 'true',
+                 'CFLAGS': '-DGPR_BACKWARDS_COMPATIBILITY_MODE'}
```

Comment:
for macos, the CFLAGS will get rewritten by CFLAGS set by macos_arch_env a few lines lower.